joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -104,6 +104,10 @@ trait AstCreatorHelper(implicit withSchemaValidation: ValidationMode) { this: As
       case StringLiteralExpression                        => BuiltinTypes.DotNetTypeMap(BuiltinTypes.String)
       case TrueLiteralExpression | FalseLiteralExpression => BuiltinTypes.DotNetTypeMap(BuiltinTypes.Bool)
       case IdentifierName                                 => typeFromTypeString(nameFromNode(node))
+      case ObjectCreationExpression =>
+        scope
```

Comment:
Could we reduce the duplication?
```scala
       val typeName = nameFromNode(createDotNetNodeInfo(node.json(ParserKeys.Type)))
        scope
           .tryResolveTypeReference(typeName)
           .getOrElse(s"${UnresolvedNamespace}.$typeName")
```

Reply:
Adding the UnresolvedNamespace part will need some test case changes, as we'll prefix some typeFullNames with this. Is it okay?